app/src/main/java/com/xmartlabs/scasas/criminalintent/ui/crime/simple/CrimeFragment.java

```diff
@@ -60,11 +60,11 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     UUID crimeId = (UUID) getArguments().getSerializable(CRIME_ID);
     ButterKnife.bind(this, view);
     if (crimeId != null) {
```

Comment:
You can set the criminal variable as a result of an optional.
Something like
```java
crime = Optional.ofNullable(crimeId)
                  .map(CrimeController.getINSTANCE()::getCrime)
                  .executeIfPresent(crime -> setValues())
                  .ofElse(Crime.builder().blabla)
```

Just a thought